instrumentation/servlet/servlet-5.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v5_0/dispatcher/RequestDispatcherAdvice.java

```diff
@@ -22,48 +23,44 @@
   public static void start(
       @Advice.Origin Method method,
       @Advice.This RequestDispatcher dispatcher,
-      @Advice.Local("_originalContext") Object originalContext,
+      @Advice.Local("otelRequestContext") Context requestContext,
       @Advice.Local("otelContext") Context context,
       @Advice.Local("otelScope") Scope scope,
       @Advice.Argument(0) ServletRequest request) {
 
-    Context parent =
-        RequestDispatcherAdviceHelper.getStartParentContext(
-            request.getAttribute(CONTEXT_ATTRIBUTE));
+    Context currentContext = Java8BytecodeBridge.currentContext();
 
-    if (parent == null) {
+    Object requestContextAttr = request.getAttribute(CONTEXT_ATTRIBUTE);
+    requestContext = requestContextAttr instanceof Context ? (Context) requestContextAttr : null;
+
+    Context parentContext =
+        RequestDispatcherAdviceHelper.getStartParentContext(currentContext, requestContext);
+    if (parentContext == null) {
       return;
     }
 
-    try (Scope ignored = parent.makeCurrent()) {
```

Comment:
This is one reason I made the decision to require `Context` as an argument to `start` in the new `Instrumenter` API. I wasn't too sure of that, since it makes writing instrumentation one step harder, but looking at this chance of messiness it seems like a good idea.